src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala

```diff
@@ -101,6 +101,138 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
   }
 
 
+  /**
+    * Helper method to decide whether an input can be spent
+    *
+    *
+    * @param validationBefore - transaction validation before checking the input
+    * @param boxesToSpend - input boxes
+    * @param dataBoxes - data input boxes
+    * @param box - the input to check
+    * @param inputIndex - index of the input to check
+    * @param stateContext - context used during validation
+    * @param currentTxCost - accumulated validation cost before checking the input
+    * @param verifier - verifier used to check spending validity
+    * @return
+    */
+  private def verifyInput(validationBefore: ValidationState[Long],
+                          boxesToSpend: IndexedSeq[ErgoBox],
+                          dataBoxes: IndexedSeq[ErgoBox],
+                          box: ErgoBox, //todo: replace with boxesToSpend(inputIndex) during next refactoring
+                          inputIndex: Short,
+                          stateContext: ErgoStateContext,
+                          currentTxCost: Long)
+                         (implicit verifier: ErgoInterpreter): ValidationResult[Long] = {
+
+    // Cost limit per block
+    val maxCost = stateContext.currentParameters.maxBlockCost
+
+    val input = inputs(inputIndex)
+
+    // Just in case, should always be true if client implementation is correct.
+    if(!box.id.sameElements(input.boxId)) {
+      log.error("Critical client error: box is not inputs(inputIndex)")
```

Comment:
I wonder what is the reason to "downgrade" this check from `validEquals` which fails the whole validation to `log.error`?

Reply:
I don't think it is safe to change the validation logic. What is the motivation?